src/app/WriteHandler.cpp

```diff
@@ -138,11 +138,17 @@ CHIP_ERROR WriteHandler::OnMessageReceived(Messaging::ExchangeContext * apExchan
                        "Incoming exchange context should be same as the initial request.");
     VerifyOrDieWithMsg(!apExchangeContext->IsGroupExchangeContext(), DataManagement,
                        "OnMessageReceived should not be called on GroupExchangeContext");
+
     if (!aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::WriteRequest))
     {
         ChipLogDetail(DataManagement, "Unexpected message type %d", aPayloadHeader.GetMessageType());
+        if (!aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
```

Comment:
Wait, why?  A StatusResponse is in fact unexpected here, right?  Why would we not respond with InvalidAction to that?

Reply:
agree, fixed, thanks